### PR TITLE
Refine pairing model

### PR DESCRIPTION
**Pull request by YodaLightsabr**
Refine receipt pairing model by fuzzysearching for the transaction memo

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -296,6 +299,9 @@ GEM
     fugit (1.8.1)
       et-orbi (~> 1, >= 1.2.7)
       raabro (~> 1.4)
+    fuzzy-string-match (1.0.1)
+      RubyInline (>= 3.8.6)
+    fuzzy_match (2.1.0)
```
Comment (YodaLightsabr) [inline Gemfile.lock]:
```suggestion
```

File: Gemfile.lock [hunk 2]
```diff
@@ -812,6 +818,8 @@ DEPENDENCIES
   flipper-ui
   foreman
   friendly_id (~> 5.2.0)
+  fuzzy-string-match (~> 1.0)
+  fuzzy_match (~> 2.1)
```
Comment (YodaLightsabr) [inline Gemfile.lock]:
```suggestion
```

File: Gemfile.lock [hunk 3]
```diff
@@ -893,4 +901,4 @@ RUBY VERSION
    ruby 3.1.4p223
 
 BUNDLED WITH
-   2.4.12
+   2.4.13
```
Comment (YodaLightsabr) [inline Gemfile.lock]:
```suggestion
   2.4.12
```

File: Gemfile.lock [hunk 4]
```diff
@@ -776,7 +779,7 @@ GEM
     zeitwerk (2.6.8)
 
 PLATFORMS
-  ruby
+  aarch64-linux
```
Comment (YodaLightsabr) [inline Gemfile.lock]:
```suggestion
  aarch64-linux
```

**Comment by garyhtou** [comment]
How does Jaro-Winkler compare to Levenshtein distance for our use case?

**Pull request closed without merging**